tck/src/test/java/org/eclipse/microprofile/jwt/tck/container/jaxrs/jwe/RolesAllowedEncryptTest.java

```diff
@@ -0,0 +1,300 @@
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Base64;
+
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.microprofile.jwt.tck.TCKConstants;
+import org.eclipse.microprofile.jwt.tck.container.jaxrs.RolesEndpoint;
+import org.eclipse.microprofile.jwt.tck.container.jaxrs.TCKApplication;
+import org.eclipse.microprofile.jwt.tck.util.MpJwtTestVersion;
+import org.eclipse.microprofile.jwt.tck.util.TokenUtils;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.Assert;
+import org.testng.Reporter;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * Tests of the MP-JWT auth method authorization behavior as expected by the MP-JWT RBAC 1.0 spec
+ */
+public class RolesAllowedEncryptTest extends Arquillian {
+
+    /**
+     * The test generated JWT token string
+     */
+    private static String token;
+
+    /**
+     * The base URL for the container under test
+     */
+    @ArquillianResource
+    private URL baseURL;
+
+    /**
+     * Create a CDI aware base web application archive
+     * @return the base base web application archive
+     * @throws IOException - on resource failure
+     */
+    @Deployment(testable=true)
+    public static WebArchive createDeployment() throws IOException {
+        URL config = RolesAllowedEncryptTest.class.getResource("/META-INF/microprofile-config-decrypt.properties");
+        URL privateKey = RolesAllowedEncryptTest.class.getResource("/privateKey.pem");
+        WebArchive webArchive = ShrinkWrap
+            .create(WebArchive.class, "RolesAllowedEncryptTest.war")
+            .addAsManifestResource(new StringAsset(MpJwtTestVersion.MPJWT_V_1_2.name()), MpJwtTestVersion.MANIFEST_NAME)
+            .addAsResource(privateKey, "/privateKey.pem")
+            .addClass(RolesEndpoint.class)
+            .addClass(TCKApplication.class)
+            .addAsWebInfResource("beans.xml", "beans.xml")
+            .addAsManifestResource(config, "microprofile-config.properties");
+        return webArchive;
+    }
+
+    @BeforeClass(alwaysRun=true)
+    public static void generateToken() throws Exception {
+        token = encryptClaims("/Token1.json");
+    }
+    
+    private static String encryptClaims(String jsonResName) throws Exception {
+        return TokenUtils.encryptClaims(jsonResName);
+    }
+
+    @RunAsClient
+    @Test(groups = TEST_GROUP_JAXRS, description = "Validate a request with no token fails with HTTP_UNAUTHORIZED")
+    public void callEchoNoAuth() throws Exception {
```

Comment:
Exception not thrown in method so can be removed.

This is the case for most test methods in the class.